libavutil/hwcontext_vulkan.c

```diff
@@ -3290,8 +3290,13 @@ static void vulkan_unmap_to_drm(AVHWFramesContext *hwfc, HWMapDescriptor *hwmap)
     av_free(drm_desc);
 }
 
-static inline uint32_t vulkan_fmt_to_drm(VkFormat vkfmt)
+static inline uint32_t vulkan_fmt_to_drm(VkFormat vkfmt,
+                                         enum AVPixelFormat sw_format)
 {
+    if (vkfmt == VK_FORMAT_R8G8_UNORM &&
+       (sw_format == AV_PIX_FMT_NV21 || sw_format == AV_PIX_FMT_NV42))
+            return DRM_FORMAT_GR88;
+
```

Comment:
How about using code below ? If so, we needn't swap the positions in the list

```
if (vkfmt == VK_FORMAT_R8G8_UNORM &&
       (sw_format == AV_PIX_FMT_NV12 || sw_format == AV_PIX_FMT_NV24))
            return DRM_FORMAT_RG88;
```

Reply:
Is a similar check needed for `VK_FORMAT_R16G16_UNORM` ?

Reply:
VK_FORMAT_R16G16_UNORM will be mapped to DRM_FORMAT_GR1616 which is wrong so I swap it's position and I swap R8G8 as well.
I see there are only P010 and P016 using VK_FORMAT_R16G16_UNORM, and DRM_FORMAT_GR1616 is not used, so I think we don't need to check R16G16.

Reply:
If so, how about to remove `{ DRM_FORMAT_GR1616,   VK_FORMAT_R16G16_UNORM   },` from this list ?